Improve domain search on Partner address domain

jaumef opened the pull request:
Improve domain search with ilike and wildcarts

- [x] Search for partners which domain is like the email's domain (Only if not found as equal).
- [x] Remove trailing characters (spaces and that sort of chars) from email domain.

ecarreras commented:
> Search for partners which domain is like the email's domain (Only if not found as equal).

In which case will apply this?

jaumef commented:
@ecarreras sometimes, the domain comes with "strange" trailing characters. I've added a strip() but if even strip does not get rid of those I've added the "partial" search.

ecarreras commented:
> I've added the "partial" search.

I don't like the "partial" search, the domain must be or not equal but not partial.
